client/ayon_core/pipeline/publish/lib.py

```diff
@@ -933,3 +934,54 @@ def get_publish_instance_families(instance):
         families.discard(family)
     output.extend(families)
     return output
+
+
+def get_instance_expected_output_path(
+        instance: pyblish.api.Instance,
+        representation_name: str,
+        ext: Optional[str] = None,
+        version: Optional[str] = None
+):
+    """Return expected publish filepath for representation in instance
+
+    This does not validate whether the instance has any representation by the
+    given name, extension and/or version.
+
+    Arguments:
+        instance (pyblish.api.Instance): publish instance
+        representation_name (str): representation name
+        ext (Optional[str]): extension for the file, useful if `name` != `ext`
+        version (Optional[int]): if provided, force it to format to this
+            particular version.
+
+    Returns:
+        str: Resolved path
+
+    """
+
+    if ext is None:
+        ext = representation_name
+    if version is None:
+        version = instance.data["version"]
+
+    context = instance.context
+    anatomy = context.data["anatomy"]
+
+    template_data = copy.deepcopy(instance.data["anatomyData"])
+    template_data.update(get_template_data_with_names(
+        project_name=context.data["projectName"],
+        folder_path=instance.data["folderPath"],
+        task_name=instance.data["task"],
+        host_name=context.data["hostName"],
+        settings=context.data["project_settings"]
+    ))
```

Comment:
Why is this needed? It should already be in `"antomyData"`.
```suggestion
```

Reply:
Good question - I'll need to debug why this was needed. I vaguely recall the data not being correct - maybe it referring to the current context instead of the instance's context. I'll start checking to see if this changes anything if removed!

Reply:
In some simple tests it did indeed seem completely redundant. If you can think of any case where this may break let me know and then I'll test it. For now, you seem right and YOLO.